pyanaconda/ui/gui/spokes/network.py

```diff
@@ -1221,6 +1123,211 @@ def _try_disconnect(obj, callback):
+
+        mode = ap.get_mode()
+        if not mode:
+            return
+
+        security = self._ap_security(ap)
+
+        ssid_str = NM.utils_ssid_to_utf8(ssid)
+
+        active = ssid == active_ssid
+
+        # the third column is for sorting
+        self._store.append([ssid, ssid_str, ssid_str, ap.get_strength(), mode, security, active])
+
+    def _ap_security(self, ap):
+        ty = NM_AP_SEC_UNKNOWN
+
+        flags = ap.get_flags()
+        wpa_flags = ap.get_wpa_flags()
+        rsn_flags = ap.get_rsn_flags()
+
+        if not flags & NM._80211ApFlags.PRIVACY \
+                and wpa_flags == NM._80211ApSecurityFlags.NONE \
+                and rsn_flags == NM._80211ApSecurityFlags.NONE:
+            ty = NM_AP_SEC_NONE
+        elif flags & NM._80211ApFlags.PRIVACY \
+                and wpa_flags == NM._80211ApSecurityFlags.NONE \
+                and rsn_flags == NM._80211ApSecurityFlags.NONE:
+            ty = NM_AP_SEC_WEP
+        elif not flags & NM._80211ApFlags.PRIVACY \
+                and wpa_flags != NM._80211ApSecurityFlags.NONE \
+                and rsn_flags != NM._80211ApSecurityFlags.NONE:
+            ty = NM_AP_SEC_WPA
+        else:
+            ty = NM_AP_SEC_WPA2
+
+        return ty
+
+    def run(self):
+        self.window.show()
+        rc = self.window.run()
+        self.window.hide()
+
+        if rc == 1:
+            ssid = self.selected_ssid
+            if ssid:
+                log.info("selected access point to be activated: %s", ssid)
+                device = self._nm_client.get_device_by_iface(self._device_name)
+                if device:
+                    self._activate_wireless_network(device, ssid)
+                else:
+                    log.warnig("device for interface %s not found", device)
+
+        return rc
+
+    def _activate_wireless_network(self, device, ssid):
+        ap = self._get_strongest_ap_for_ssid(device.get_access_points(), ssid)
+        if not ap:
+            return
+
+        cons = ap.filter_connections(device.filter_connections(self._nm_client.get_connections()))
+        if cons:
+            con = cons[0]
+            self._nm_client.activate_connection_async(con, device, ap.get_path(), None)
+        else:
+            if self._ap_is_enterprise(ap):
+                # Create a connection for the ap and [Configure] it later with nm-c-e
+                ssid_str = NM.utils_ssid_to_utf8(ssid)
+                con = NM.SimpleConnection.new()
+                s_con = NM.SettingConnection.new()
+                s_con.set_property('uuid', str(uuid4()))
+                s_con.set_property('id', ssid_str)
+                s_con.set_property('type', NM_CONNECTION_TYPE_WIFI)
+                s_wireless = NM.SettingWireless.new()
+                s_wireless.set_property('ssid', ap.get_ssid())
+                s_wireless.set_property('mode', 'infrastructure')
+                con.add_setting(s_con)
+                con.add_setting(s_wireless)
```

Comment:
It could be interesting to have these configurations and other parts of ``NM`` abstracted. Could be pretty similar to our DBus structure solution.

Reply:
The NM settings are dict of dicts for extensibility. I don't see a point in creating the abstraction at this point, but maybe in the future I'll see it. In any case I wouldn't make it part of this PR.